pkg/messageFormat/ast.go

```diff
@@ -0,0 +1,101 @@
+package messageFormat
+
+import (
+	"fmt"
+	"strings"
+)
+
+type NodeMatch struct {
```

Comment:
Note: that types that are not used or planned to be used outside the package shouldn't be exposed.